### PR TITLE
feat(headless): send clientId instead of visitorId in case assist

### PR DESCRIPTION
**Pull request by nathanlb**
The customer service api now accepts `clientId` or `visitorId`.
We want to favour `clientId` to phase out the other so after this PR the case assist client will send that instead.

### CODE REVIEW

File: packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
 
 export type GetCaseClassificationsRequest = BaseParam &
   CaseAssistIdParam &
-  VisitorIDParam &
+  ClientIDParam &
```
Comment (lbergeron) [inline packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts:17]: Isn't removing `VisitorIDParam` from the type a breaking change? An error would be raised for the snippet below.

```typescript
buildGetCaseClassificationsRequest({ ..., visitorId: 'some-id' });
```

**Review by lbergeron — COMMENTED** [review]
I don't think we can safely remove `VisitorIDParam` from the request types.

File: packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts (hunk 1 above)
Comment (nathanlb) [reply packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts:17]: Given the client is not exported I don't think there can be any instance of such usage outside of Headless. At least I couldn't find any hit of it.

Comment (nathanlb) [reply packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts:17]: I could easily leave it in though if we do find it is breaking.

Comment (lbergeron) [reply packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts:17]: > Given the client is not exported I don't think there can be any instance of such usage outside of Headless.

If it is not exported by the Headless bundle, then we're good.

**Review by lbergeron — APPROVED** [review]
It turns out the modified request types are not exposed by Headless so removing `VisitorIDParam` from the type is not an issue.

**Review by mmitiche — APPROVED** [review]
LGTM!

File: packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts (hunk 1 above)
Comment (dmbrooke) [reply packages/headless/src/api/service/case-assist/get-case-classifications/get-case-classifications-request.ts:17]: As far as I can tell, this is only used within Headless itself, so it should be okay 🙂

**Review by dmbrooke — APPROVED** [review]

**Pull request merged**